astropy/units/physical.py

```diff
@@ -1,133 +1,458 @@
+        self._physical_type_list = sorted(self._physical_type)
+
+    def __iter__(self):
+        yield from self._physical_type_list
+
+    def __eq__(self, other):
+        """
+        Return `True` if ``other`` represents a physical type that is
+        consistent with the physical type of the `PhysicalType` instance.
+        """
+        if isinstance(other, PhysicalType):
+            return self._physical_type_id == other._physical_type_id
+        elif isinstance(other, str):
+            other = _standardize_physical_type_names(other)
+            return other.issubset(self._physical_type)
+        else:
+            return NotImplemented
+
+    def __ne__(self, other):
+        equality = self.__eq__(other)
+        return not equality if isinstance(equality, bool) else NotImplemented
+
+    @property
+    def _name_string_as_ordered_set(self):
+        return "{" + str(self._physical_type_list)[1:-1] + "}"
+
+    def __repr__(self):
+        if len(self._physical_type) == 1:
+            names = "'" + self._physical_type_list[0] + "'"
+        else:
+            names = self._name_string_as_ordered_set
+        return f"PhysicalType({names})"
+
+    def __str__(self):
+        if len(self._physical_type) == 1:
+            return self._physical_type_list[0]
+        else:
+            return self._name_string_as_ordered_set
+
+    @staticmethod
+    def _identify_unit_from_unit_or_physical_type(obj):
+        """
+        If a unit is passed in, return that unit.  If a physical type is
+        passed in, return a unit that corresponds to that physical type.
+        Otherwise, return `NotImplemented`.
+        """
+        if isinstance(obj, core.UnitBase):
+            return _replace_temperatures_with_kelvin(obj)
+        elif isinstance(obj, PhysicalType):
+            return obj._unit
+        else:
+            return NotImplemented
+
+    def _dimensional_analysis(self, other, operation):
+        other_unit = self._identify_unit_from_unit_or_physical_type(other)
+        if other_unit is NotImplemented:
+            return NotImplemented
+        other_unit = _replace_temperatures_with_kelvin(other_unit)
+        new_unit = getattr(self._unit, operation)(other_unit)
+        return new_unit.physical_type
+
+    def __mul__(self, other):
+        return self._dimensional_analysis(other, "__mul__")
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    def __truediv__(self, other):
+        return self._dimensional_analysis(other, "__truediv__")
+
+    def __rtruediv__(self, other):
+        other = self._identify_unit_from_unit_or_physical_type(other)
+        if other is NotImplemented:
+            return NotImplemented
+        return other.physical_type._dimensional_analysis(self, "__truediv__")
+
+    def __pow__(self, power):
+        return (self._unit ** power).physical_type
+
+    def __hash__(self):
+        return hash(self._physical_type_id)
```

Comment:
I might or might not be looking forward to using physical type instances as keys in dictionaries. :smile_cat:

Reply:
Nice!